internal/service/identitystore/group_test.go

```diff
@@ -0,0 +1,173 @@
+			testAccPreCheck(t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, names.IdentityStoreEndpointID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGroupConfig_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGroupExists(resourceName, &group),
+					acctest.CheckResourceDisappears(acctest.Provider, tfidentitystore.ResourceGroup(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckGroupDestroy(s *terraform.State) error {
+	conn := acctest.Provider.Meta().(*conns.AWSClient).IdentityStoreConn
+	ctx := context.Background()
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_identitystore_group" {
+			continue
+		}
+
+		_, err := conn.DescribeGroup(ctx, &identitystore.DescribeGroupInput{
+			GroupId:         aws.String(rs.Primary.Attributes["group_id"]),
+			IdentityStoreId: aws.String(rs.Primary.Attributes["identity_store_id"]),
+		})
+		if err != nil {
+			var nfe *types.ResourceNotFoundException
+			if errors.As(err, &nfe) {
+				return nil
+			}
+			return err
+		}
+
+		return create.Error(names.IdentityStore, create.ErrActionCheckingDestroyed, tfidentitystore.ResNameGroup, rs.Primary.ID, errors.New("not destroyed"))
+	}
+
+	return nil
+}
+
+func testAccCheckGroupExists(name string, group *identitystore.DescribeGroupOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return create.Error(names.IdentityStore, create.ErrActionCheckingExistence, tfidentitystore.ResNameGroup, name, errors.New("not found"))
+		}
+
+		if rs.Primary.ID == "" {
+			return create.Error(names.IdentityStore, create.ErrActionCheckingExistence, tfidentitystore.ResNameGroup, name, errors.New("not set"))
+		}
+
+		conn := acctest.Provider.Meta().(*conns.AWSClient).IdentityStoreConn
+		ctx := context.Background()
+		resp, err := conn.DescribeGroup(ctx, &identitystore.DescribeGroupInput{
+			GroupId:         aws.String(rs.Primary.Attributes["group_id"]),
+			IdentityStoreId: aws.String(rs.Primary.Attributes["identity_store_id"]),
+		})
+
+		if err != nil {
+			return create.Error(names.IdentityStore, create.ErrActionCheckingExistence, tfidentitystore.ResNameGroup, rs.Primary.ID, err)
+		}
+
+		*group = *resp
+
+		return nil
+	}
+}
+
+func testAccPreCheck(t *testing.T) {
+	//conn := acctest.Provider.Meta().(*conns.AWSClient).IdentityStoreConn
+	//ctx := context.Background()
+	//
+	//input := &identitystore.ListGroupsInput{}
+	//_, err := conn.ListGroups(ctx, input)
+	//
+	//if acctest.PreCheckSkipError(err) {
+	//	t.Skipf("skipping acceptance testing: %s", err)
+	//}
+	//
+	//if err != nil {
+	//	t.Fatalf("unexpected PreCheck error: %s", err)
+	//}
+}
+
+func testAccCheckGroupNotRecreated(before, after *identitystore.DescribeGroupOutput) resource.TestCheckFunc {
```

Comment:
Note to self: `testAccCheckGroupNotRecreated` can be removed since it's not used.